src/libmongoc/tests/test-libmongoc.c

```diff
@@ -1132,10 +1132,20 @@ call_ismaster_with_host_and_port (char *host_and_port, bson_t *reply)
 #endif
 
    if (!mongoc_client_command_simple (
-          client, "admin", tmp_bson ("{'isMaster': 1}"), NULL, reply, &error)) {
-      fprintf (stderr, "error calling ismaster: '%s'\n", error.message);
-      fprintf (stderr, "URI = %s\n", uri_str);
-      abort ();
+          client, "admin", tmp_bson ("{'hello': 1}"), NULL, reply, &error)) {
+      bson_destroy (reply);
+
+      if (!mongoc_client_command_simple (
+             client,
+             "admin",
+             tmp_bson ("{'" HANDSHAKE_CMD_LEGACY_HELLO "': 1}"),
+             NULL,
+             reply,
+             &error)) {
+         fprintf (stderr, "error calling hello: '%s'\n", error.message);
+         fprintf (stderr, "URI = %s\n", uri_str);
+         abort ();
+      }
```

Comment:
I opted for manually calling `hello` first and then falling back to legacy hello if the server does not support it. Other tests still use legacy hello to avoid failing tests on older server versions that don't support `hello`. We could wrap this in a `test_mongoc_client_command_hello` function and use that exclusively. Please let me know if you'd like me to add that.

Reply:
Nice, I don't feel strongly either way about switching other tests. But perhaps it will be necessary anyway with versioned API testing to use `hello` everywhere?

Reply:
Yes, there's going to be follow-up work once DRIVERS-1293 is ready for implementation. I'll finish this change then.